scripts/deploy/gce/node

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env bash
+
+set -eoux pipefail
+
+#
+# Deploy a full node on GCE
+#
+# See ./bootstrap for assumptions about the environment.
+#
+# Note: the 'gcloud' beta commands have to be installed (`gcloud components
+# install beta`).
+#
+
+DRY_RUN=${DRY_RUN:-""}
+
+OSCOIN_VERSION=${OSCOIN_VERSION:-$(git rev-parse HEAD)}
+OSCOIN_NETWORK=${OSCOIN_NETWORK:?Network not set}
+OSCOIN_REPLICAS=${OSCOIN_REPLICAS:-3}
+OSCOIN_GOSSIP_PORT=${OSCOIN_GOSSIP_PORT:-6942}
+OSCOIN_API_PORT=${OSCOIN_API_PORT:-8080}
+OSCOIN_METRICS_PORT=${OSCOIN_METRICS_PORT:-8081}
+OSCOIN_BENEFICIARY=${OSCOIN_BENEFICIARY:?Beneficiary address not set}
+
+GCE_PROJECT_ID=${GCE_PROJECT_ID:-opensourcecoin}
+GCE_REGION=${GCE_REGION:-europe-west1}
+GCE_VPC=${GCE_VPC:-$OSCOIN_NETWORK}
+GCE_SERVICEACCOUNT=${GCE_SERVICEACCOUNT:-oscoin-node@opensourcecoin.iam.gserviceaccount.com}
+GCE_MACHINE_TYPE=${GCE_MACHINE_TYPE:-n1-standard-1}
+GCE_DNS_ZONE=${GCE_DNS_ZONE:-oscoin-internal}
+GCE_DNS_DOMAIN=${GCE_DNS_DOMAIN:-oscoin.internal}
+GCE_NUM_SSDS=${GCE_NUM_SSDS:-1};
+GCE_NUM_SSDS=$((GCE_NUM_SSDS > 8 ? 8 : (GCE_NUM_SSDS < 1 ? 1 : GCE_NUM_SSDS)))
+GCE_UPDATE_MIN_READY=${GCE_UPDATE_MIN_READY:-t2m}
+
+OSCOIN_DEPLOY=${OSCOIN_DEPLOY:-cabal v2-run -- oscoin-deploy}
+
+# create machine config via cloud-init
+_clowninit=$(mktemp "/tmp/node.cloudinit.yaml.XXXXX")
+$OSCOIN_DEPLOY node \
+    --format=cloud-init \
+    --version="$OSCOIN_VERSION" \
+    --network="$OSCOIN_NETWORK" \
+    --host="0.0.0.0" \
```

Comment:
I think that if we want to execute on our plan of syncing via the HTTP API and by using the (handshaked) `NodeInfo`, we would need the (real) external IP of the machine here, otherwise all the peers would advertise something like `Addr { host = "0.0.0.0", port = 8080 }`. If you think this is not feasible, perhaps we should extend the handshake routine and _not_ use the IP address passed via the CLI but read it (effectfully) when the application starts?

Reply:
The "self" address is [resolved during startup](https://github.com/oscoin/oscoin/blob/master/src/Oscoin/P2P.hs#L133). Am I missing something?

Reply:
I think perhaps I am the one missing something, as I think I have (in my previous PR) passed to the gossip the CLI address rather than the `self` one 🙈 : 

```haskell
            let myHost     = P2P.numericHost optHost
            let myApiInfo  = P2P.mkAddr myHost optApiPort
            let myNodeInfo = P2P.mkNodeInfo myApiInfo nid
            let mySK       = snd keys
            gossip <- managed $
                withGossip telemetry
                           P2P.BootstrapInfo
                               { P2P.bootNodeId      = pure myNodeInfo
                               , P2P.bootHttpApiAddr = pure myApiInfo
                               , P2P.bootGossipAddr  = P2P.mkAddr myHost optGossipPort
                               }
                           (Set.map (Nothing,) <$> disco)
                           (storage node)
                           (Handshake.secureHandshake
                               mySK
                               myNodeInfo
                               (P2P.Disco.optNetwork optDiscovery'))
```